benchmark/benchmark_tridiagonal_solver.cpp

```diff
@@ -0,0 +1,118 @@
+
+#include <tuvx/linear_algebra/linear_algebra.hpp>
+
+#include <benchmark/benchmark.h>
+
+#ifdef TUVX_COMPILE_WITH_INTEL
+  #include <mkl_lapacke.h>
+#elif TUVX_COMPILE_WITH_GCC
+  #include <lapacke.h>
+#endif
+
+using namespace tuvx;
+typedef TridiagonalMatrix<double> trid_matd;
+typedef std::vector<double> vecd;
+
+typedef TridiagonalMatrix<float> trid_matf;
+typedef std::vector<float> vecf;
```

Comment:
I would recommend using the underlying types directly instead of adding a `typedef`. It makes it clearer in the code what these actually are.

Reply:
+1

Reply:
Sounds good @mattldawson , I will make this change in my next commit